likecoin/js/admin/likecoin_metabox.js

```diff
@@ -18,61 +41,186 @@ async function onRefreshPublishStatus(e) {
+  const isWordpressPublished = res.wordpress_published;
+  lcPostInfo.isMattersPublished = res.matters.status;
+  if (iscnHash && iscnId) { // state done
     const iscnIdString = encodeURIComponent(iscnId);
-    lcPostInfo.mainStatusTitle = 'LIVE on #DePub';
     mainStatusField.textContent = '';
-    mainStatusField.innerHTML = `<h3>${lcPostInfo.mainStatusTitle}</h3>`;
-    arweaveTextField.innerHTML = `<a rel="noopener" target="_blank" href="https://app.like.co/view/${iscnIdString}">${iscnId}</a>`;
-  } else if (wordpressPublished === 'publish' && (arweaveAndISCNUploadStatus === 'initial' || arweaveAndISCNUploadStatus === 'failed')) {
-    lcPostInfo.mainStatusTitle = 'READY to #DePub';
+    const statusDot = createElementWithAttrbutes('h1', {
+      text: ' · ',
+      className: 'iscn-status-green',
+    });
+    const statusText = createElementWithAttrbutes('h3', {
+      text: lcPostInfo.mainTitleDone,
+      className: 'iscn-status-text',
+    });
+    mainStatusField.appendChild(statusDot);
+    mainStatusField.appendChild(statusText);
+    const ISCNLink = createElementWithAttrbutes('a', {
+      text: iscnId,
+      rel: 'noopener',
+      target: '_blank',
+      href: `https://app.like.co/view/${iscnIdString}`,
+    });
+    ISCNStatusTextField.textContent = '';
+    ISCNStatusTextField.appendChild(ISCNLink);
+  } else if ( // show button
+    isWordpressPublished === 'publish'
+    && (lcPostInfo.mainStatus === 'initial' || lcPostInfo.mainStatus === 'failed')
+  ) {
     mainStatusField.textContent = '';
-    mainStatusField.innerHTML = `<h3>${lcPostInfo.mainStatusTitle}</h3>`;
-    const uploadArweaveBtn = document.createElement('button');
-    uploadArweaveBtn.innerText = 'Submit to ISCN';
-    uploadArweaveBtn.setAttribute('id', 'lcArweaveUploadBtn');
-    uploadArweaveBtn.setAttribute('class', 'button button-primary');
-    arweaveTextField.textContent = '';
-    arweaveTextField.appendChild(uploadArweaveBtn);
-    uploadArweaveBtn.addEventListener('click', onEstimateAndUploadArweave);
-  } else if (wordpressPublished !== 'publish') {
-    arweaveTextField.innerHTML = '-';
-  } else { // post published but upload arweave is still in intermediate steps
-    arweaveTextField.innerHTML = `${arweaveAndISCNUploadStatus}`;
-  }
-  if (matters.url) {
-    const { url, status } = matters;
-    mattersTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
+    const statusDot = createElementWithAttrbutes('h1', {
+      text: ' · ',
+      className: 'iscn-status-orange',
+    });
+    const statusText = createElementWithAttrbutes('h3', {
+      text: lcPostInfo.mainTitleIntermediate,
+      className: 'iscn-status-text',
+    });
+    mainStatusField.appendChild(statusDot);
+    mainStatusField.appendChild(statusText);
+    const arweaveISCNBtn = createElementWithAttrbutes('button', {
+      text: 'Submit to ISCN',
+      className: 'button button-primary',
+      id: 'lcArweaveUploadBtn',
+    });
```

Comment:
If you always need to do these steps for updating `mainStatusField`, these should be refactor to a function too